api/src/auth.js

```diff
@@ -63,6 +63,11 @@ module.exports = {
            hasRole(userCtx, ONLINE_ROLE);
   },
 
+  isOffline: roles => {
+    const configured = config.get('roles') || {};
+    return roles.some(role => configured[role] && configured[role].offline);
+  },
```

Comment:
It'd be good to have some code reuse between this and `isOnlineOnly` - they do the exact opposite thing I think?

Maybe change `isOnlineOnly` to take an array of roles?

Reply:
Well, they don't do the exact opposite thing. 
`isOnlineOnly` checks for the `mm-online` flag-role in a provided role list part of a `userCtx`, regardless of roles config, while `isOffline` checks config to see if the provided roles are configured as offline. 
Do you think it would be more correct for `isOnlineOnly` to check config - and the `mm-online` flag to only be useful in `webapp` while bootstrapping?

Reply:
> Do you think it would be more correct for isOnlineOnly to check config - and the mm-online flag to only be useful in webapp while bootstrapping?

Yes I think so. That way they're consistent as much as possible.

Reply:
I had a bit of a philosophical dilemma about this: if a user had no configured role, is it online or offline? One can easily get to this point if a role is created, a user with this role is created, role is removed. 
I decided that one such user should be considered offline and updated the `isOffline` function to return true if none of the provided roles are configured, along with doing a `!isOffline` in the `isOnlineOnly`. But I'm not convinced this is correct.

Reply:
I think in that case it should fail before it gets here, ie: every request should 403 if you don't have any known role, because you're not permitted to do anything! Feel free to go with what you have this time, and maybe raise an issue for that edge case?